cmd/buildah/config.go

```diff
@@ -244,7 +244,11 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		builder.SetDomainname(c.String("domainname"))
 	}
 	if c.IsSet("hostname") {
-		builder.SetHostname(c.String("hostname"))
+		name := c.String("hostname")
+		if name != "" && builder.Format != buildah.Dockerv2ImageManifest {
```

Comment:
Can we instead use `... != buildah.OCIv1ImageManifest`? Currently, it works as there are only two formats but who knows what the future holds?

Reply:
So you want to only report the warning, if buildah.Format == buildah.OCIv1ImageManifest, since we know that OCIv1ImageManifest does not support it but OCIV2ImageManifest, might so we would not need to change the code.